### PR TITLE
chore: bump vyper version

### PR DESCRIPTION
**Pull request by steffenix**

### CODE REVIEW

**Comment by fubuloubu** [comment]
Do we want to update the 0.4.0 tag for this?

**Review by fubuloubu — APPROVED** [review]

File: contracts/Registry.vy [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# @version 0.2.11
+# @version 0.2.12
```
Comment (fubuloubu) [inline contracts/Registry.vy:1]: Can't update the Registry

**Review by fubuloubu — CHANGES_REQUESTED** [review]

**Comment by steffenix** [comment]
> Do we want to update the 0.4.0 tag for this?

I am thinking we should tag 0.4 with that

**Comment by fubuloubu** [comment]
> > Do we want to update the 0.4.0 tag for this?
> 
> I am thinking we should tag 0.4 with that

Let's do it!

Just not the Registry lol

**Comment by steffenix** [comment]
> > > Do we want to update the 0.4.0 tag for this?
> > 
> > 
> > I am thinking we should tag 0.4 with that
> 
> Let's do it!
> 
> Just not the Registry lol

What's about Registry? Just don't need to redeploy it or something will breek?

**Comment by fubuloubu** [comment]
> What's about Registry? Just don't need to redeploy it or something will breek?

Upgrading the Registry requires replaying all of the events in the previous version of the registry, as well as coordinating upgrades across all AffiliateWrappers and yTokens (which aren't deployed yet)

**Comment by steffenix** [comment]
I do not understand why we can't bump the Registry version we just don't deploy it again.

**Comment by fubuloubu** [comment]
> I do not understand why we can't bump the Registry version we just don't deploy it again.

We can deploy it again, its just the deployment process is much more involved than with a new `Vault` API version. There are many downstrram users of the registry that would need to be updated, and a large amount of transactions that need to be replayed on the newly deployed registry (20+ transactions)

Its possible, but not very simple, and not really worth it just to upgrade the compiler version, unless a critical bug were found or there was a new feature we needed.

**Comment by steffenix** [comment]
@fubuloubu I have made the update

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**